mineinabyss-features/src/main/kotlin/com/mineinabyss/enchants/enchantments/MagnetismListener.kt

```diff
@@ -0,0 +1,39 @@
+package com.mineinabyss.enchants.enchantments
+
+import com.mineinabyss.enchants.CustomEnchants.MAGNETISM
+import com.mineinabyss.mineinabyss.core.mineInAbyss
+import com.okkero.skedule.schedule
+import org.bukkit.entity.Item
+import org.bukkit.event.EventHandler
+import org.bukkit.event.Listener
+import org.bukkit.event.block.BlockBreakEvent
+
+
+class MagnetismListener : Listener {
+    @EventHandler
+    fun BlockBreakEvent.onBlockBreak() {
+
+        val item = player.inventory.itemInMainHand
+
+        if (item.containsEnchantment(MAGNETISM)){
+
+            val loc = block.location
+
+            mineInAbyss.schedule {
+                waitFor(1)
+
+                val entities = player.world.getNearbyEntities(loc, 4.0, 4.0, 4.0)
+
+                for (entity in entities) {
+                    if (entity is Item) {
+                        val item: Item = entity
+                        val stack = player.inventory.addItem(item.itemStack)
+                        if (stack[0] == null){
```

Comment:
If this isn't null, we still need to update the item on the ground to be the leftover amount. Think about a 64 stack of stone that we add to our inventory, but 10 remain. If we don't update the item on ground to 10, we would have 54 extra items!

Reply:
i dont see y you say that as i did some test and if there is 64 item to pick up from the ground but i can only hold 54 more there will be only 10 left on the ground :/

Reply:
Sure, ping me on discord if you'd like. If it's working, it might be working because `item.itemStack` gives the real ItemStack and not a copy, but that would be odd. I'd like to make sure ;p